pkg/catalog/inbound_traffic_policies.go

```diff
@@ -244,13 +244,20 @@ func (mc *MeshCatalog) getUpstreamServicesIncludeApex(upstreamServices []service
 		}
 
 		for _, split := range mc.meshSpec.ListTrafficSplits(smi.WithTrafficSplitBackendService(svc)) {
+			svcName := k8s.GetServiceFromHostname(split.Spec.Service)
```

Comment:
This leaks k8s primitives into the catalog. We already do this in a bunch of places, but I think we should avoid it going forward.

What do you think about adding this method to an interface (service.Provider?), and calling that? Same for GetSubdomainFromHostname

Reply:
Oooh good catch!! I agree, we shouldn't leak this 👍🏾

Reply:
FWIW a refactor like this, where we're already following this pattern a lot, I think we *can* do in a later PR :)

Reply:
Yeah, this is probably going to be a bigger de-coupling initiative that we should capture in a standalone issue. Would you mind creating that @steeling? I think you've got some good context to share that may help prioritize it